src/modules/services/memcached.nix

```diff
@@ -49,6 +49,24 @@ in
   };
 
   config = lib.mkIf cfg.enable {
-    processes.memcached.exec = "${cfg.package}/bin/memcached --port=${toString cfg.port} --listen=${cfg.bind} ${lib.concatStringsSep " " cfg.startArgs}";
+    processes.memcached = {
+      exec = "${cfg.package}/bin/memcached --port=${toString cfg.port} --listen=${cfg.bind} ${lib.concatStringsSep " " cfg.startArgs}";
+
+      process-compose = {
+        readiness_probe = {
+          exec.command = ''
+            ${pkgs.bash}/bin/bash -c 'exec 3<>/dev/tcp/${cfg.bind}/${toString cfg.port}; printf "stats\nquit\n" >&3; cat <&3'
```

Comment:
On *nix-es process-compose already executes the commands via bash, so no need for the extra process.